### PR TITLE
Enable python 3.9 again

### LINKED ISSUE #241: update orchestrator-core python to >= 3.9
The lower bound for the python version supported by orchestrator-core should match the python3 version in the latest stable release of Debian. This ensures orchestrator consumers who don't want to (or can't) use containers will not have to compile a newer version of Python.

As of today (April 2023) the latest Debian is 11 ("Bullseye") which supports python 3.9: https://packages.debian.org/stable/python3

* [x] Update orchestrator-core to support at least python 3.9
* [x] Enable 3.9 in orchestrator-core CI tests
* [x] Add this policy to the documentation

### PR DESCRIPTION
**Pull request by Mark90**
* Add python 3.9 support again, include 3.9 in CI tests
* Document python version support policy

# TODO
* [x] Release and pin [nwa-stdlib](https://github.com/workfloworchestrator/nwa-stdlib/pull/16) and [oauth2-lib](https://github.com/workfloworchestrator/oauth2-lib/pull/36)

Closes #241

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -65,14 +66,16 @@ dependencies = [
     "SQLAlchemy-Utils==0.40.0",
     "typer==0.7.0",
     "uvicorn[standard]~=0.20.0",
-    "nwa-stdlib~=1.4.6",
-    "oauth2-lib~=1.2.5",
+#    "nwa-stdlib~=1.4.6",
+    "nwa-stdlib @ git+https://github.com/workfloworchestrator/nwa-stdlib@python39-compatibility#egg=nwa-stdlib",
```
Comment (Mark90) [inline pyproject.toml:70]: https://github.com/workfloworchestrator/nwa-stdlib/pull/16

File: pyproject.toml [hunk 2]
```diff
@@ -65,14 +66,16 @@ dependencies = [
...
+#    "oauth2-lib~=1.2.5",
+    "oauth2-lib @ git+https://github.com/workfloworchestrator/oauth2-lib@python39-compatibility#egg=oauth2-lib",
```
Comment (Mark90) [inline pyproject.toml:72]: https://github.com/workfloworchestrator/oauth2-lib/pull/36

**Comment by tjeerddie** [comment]
You should check the PR that removes the 3.9 support, I see for example:
```
# This is needed to avoid having a maintenance version of nwa-stdlib for py<3.10
if sys.version_info >= (3, 10):
    from nwastdlib.logging import ClearStructlogContextASGIMiddleware
```

https://github.com/workfloworchestrator/orchestrator-core/pull/215/files#diff-f9101b3f0c19dfb5a4e53172c74c13634489a593d88031b6a8a06811fdb4f7de

File: pyproject.toml [hunk 3]
```diff
@@ -65,14 +66,16 @@ dependencies = [
...
     "markupsafe==2.0.1",
     "bandit==1.7.2",
     "tabulate==0.9.0"
 ]
 description-file = "README.md"
-requires-python = ">3.9,<3.12"
+requires-python = ">3.8,<3.12"
```
Comment (tjeerddie) [inline pyproject.toml:78]: I would prefer showing the required python version then above a previous version
```suggestion
requires-python = ">=3.9,<3.12"
```

Comment (Mark90) [reply pyproject.toml]: :+1: done

**Review by tjeerddie — APPROVED** [review]

**Comment by dmarjenburgh** [comment]
Looks good

**Review by dmarjenburgh — APPROVED** [review]
Looks

**Pull request merged**